restify/restify.d.ts

```diff
@@ -470,12 +470,12 @@ declare module "restify" {
   }
 
   interface Client {
-    get: (path: string, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    head: (path: string, callback?: (err: any, req: Request, res: Response) => any) => any;
-    post: (path: string, object: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    put: (path: string, object: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    patch: (path: string, object: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
-    del: (path: string, callback?: (err: any, req: Request, res: Response) => any) => any;
+    get: (opts: any, callback?: (err: any, req: Request, res: Response, obj: any) => any) => any;
```

Comment:
Updated the PR with the suggestion. The Client and HttpClient are so similar, but still have differences in how much they specify. I'm hesitant to make too many changes, as it might break things for people.

For example: `opts` or `opts?`, and the whole callback being just a function or a function with specific parameters.  Should they be made to look similar?